docs/user/docker_config.rst

```diff
@@ -0,0 +1,158 @@
+system that is working smoothly. The Docker Compose reference_ explains the
+format of the file, for details on syntax you are encouraged to check the
+official documentation.
+
+In the remainder of this section we will go over our example. The first line
+defines the version of the docker-compose file format, which corresponds to the
+Docker Engine version you are running. You likely do not have to change this.
+
+Every block below ``services:`` defines a Docker container. Let's look at one
+example container configuration. This example does not represent something we
+would want to actually use, but contains configuration lines relevant to many
+other container configurations::
+
+  example-container-name:
+    image: simonsobs/example-docker-image:latest
+    restart: always
+    hostname: ocs-docker
+    user: "9000"
+    ports:
+      - "127.0.0.1:8001:8001" # expose for OCS
+    volumes:
+      - /data:/data:ro
+      - ./.crossbar:/app/.crossbar
+    environment:
+        MAX_POINTS: 1000
+        SQL_HOST: "database"
+        SQL_DB: "files"
+    depends_on:
+      - "sisock-crossbar"
+      - "database"
+
+The top line, ``example-container-name``, defines the name of the service to
+docker-compose. These must be unique. ``image`` defines the docker image used
+for the container. A container can be thought of as a copy of an image. The
+container is what actually runs when you startup your docker service.
+``restart`` allows you to define when a container can be automatically
+restarted, in this instance, always. ``hostname`` defines the hostname internal
+to the container. This is used in the OCS container configurations in
+conjunction with the ocs-site-configs file. We recommend appending "-docker" to
+the hostname to distinguish Agents running within containers from those running
+directly on the host. ``user`` defines the user used inside the container. This
+is only used on the aggregator agent configuration.
+
+``ports`` defines the ports exposed from the container to the host. This is
+used on containers like the crossbar container and the grafana container.
+``volumes`` defines mounted docker volumes and bind mounts to the host system.
+The syntax here is ``/host/system/path:/container/system/path``. Alternatively
+the host system path can be a named docker volume, in which case docker manages
+the storage. ``environment`` defines environment variables inside the
+container. This is used for configuring behaviors inside the containers.
+``depends_on`` means Docker Compose will wait for the listed containers to
+start before starting this container. This does not mean the services will be
+ready, but the container will be started.
+
+.. note::
+    Environment variables can be used within a docker-compose configuration
+    file. This is done for the `OCS_CONFIG_DIR` mount for the OCS agents in the
+    default template.  For more information see the `docker compose
+    documentation`_.
```

Comment:
Could mention here that you need to run `sudo -E` if you are not part of the docker group.

Reply:
Great suggestion! I'm always forgetting to throw that flag, even in documentation apparently! Added.